File: typescript/docs_code_snippets/quickstart-streaming.ts
import { Opper } from "opperai";

const opper = new Opper();

for await (const chunk of opper.stream("respond", {
  instructions: "Respond in Swedish",
  input: "What are the benefits of using an AI gateway?",
})) {
  if (chunk.type === "content") process.stdout.write(chunk.delta);
  if (chunk.type === "done") console.log();
}

File: python/docs_code_snippets/quickstart-streaming.py
import sys
from opperai import Opper

opper = Opper()

# --- docs ---
for chunk in opper.stream(
    "respond",
    instructions="Respond in Swedish",
    input="What are the benefits of using an AI gateway?",
):
    if chunk.type == "content":
        sys.stdout.write(chunk.delta)
        sys.stdout.flush()
    elif chunk.type == "done":
        print()
# --- /docs ---
